injectors/anvil-injector/src/main/kotlin/io/mcarle/konvert/injector/anvil/AnvilInjector.kt

```diff
@@ -0,0 +1,89 @@
+package io.mcarle.konvert.injector.anvil
+
+import com.google.auto.service.AutoService
+import com.google.devtools.ksp.KspExperimental
+import com.google.devtools.ksp.isAnnotationPresent
+import com.google.devtools.ksp.symbol.KSClassDeclaration
+import com.squareup.anvil.annotations.ContributesBinding
+import com.squareup.anvil.annotations.ContributesMultibinding
+import com.squareup.kotlinpoet.AnnotationSpec
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.TypeSpec
+import com.squareup.kotlinpoet.ksp.toAnnotationSpec
+import dagger.MapKey
+import io.mcarle.konvert.converter.api.classDeclaration
+import io.mcarle.konvert.converter.api.config.Configuration
+import io.mcarle.konvert.injector.anvil.config.InjectionMethod
+import io.mcarle.konvert.injector.anvil.config.defaultInjectionMethod
+import io.mcarle.konvert.injector.anvil.config.defaultScope
+import io.mcarle.konvert.plugin.api.KonverterInjector
+import io.mcarle.konvert.plugin.api.passthroughAnnotation
+import javax.inject.Inject
+import javax.inject.Qualifier
+import javax.inject.Singleton
+
+@AutoService(KonverterInjector::class)
+class AnvilInjector : KonverterInjector {
+    override fun processType(builder: TypeSpec.Builder, originKSClassDeclaration: KSClassDeclaration) {
+        val defaultInjectionApplied = applyDefaults(builder)
+
+        val bindingUsed = passthroughAnnotation<KContributesBinding, ContributesBinding>(builder, originKSClassDeclaration) {
+            it.value
+        }
+        val multibindingUsed = passthroughAnnotation<KContributesMultibinding, ContributesMultibinding>(builder, originKSClassDeclaration) {
+            it.value
+        }
+        passthroughAnnotation<KSingleton, Singleton>(builder, originKSClassDeclaration) {
+            it.value
+        }
+
+        handleQualifiersAndMapKeys(builder, originKSClassDeclaration)
+
+        if (defaultInjectionApplied || bindingUsed || multibindingUsed) {
+            // We also have to add empty constructor with @Inject annotation
+            builder.addFunction(FunSpec.constructorBuilder()
+                .addAnnotation(Inject::class)
+                .build()
```

Comment:
@mcarleio I'm creating a constructor for mapper here. This might be a problem in the future if you at any point will create another constructor. Or another plugin adds it. It should be probably handled better. Any ideas?

Reply:
I see what you mean. I have to look at it later. Maybe kotlin-poet already offers something, otherwise writing a small class handling added classes, methods, etc. and pass that in addition to the injector could be a solution. But I think this is OK for now.

Reply:
I think it is already possible with kotlin poet:

```kotlin
val emptyConstructor = builder.funSpecs.firstOrNull {
    it.isConstructor && it.parameters.isEmpty()
}
if (emptyConstructor != null) {
  builder.funSpecs += emptyConstructor.toBuilder().addAnnotation(Inject::class).build()
  builder.funSpecs -= emptyConstructor
}
```

Reply:
Are you ok with leaving responsibility for that to injector? IMO it should be fine, i can adjust code for that

Reply:
Yes, I think that would be fine for now.